src/options/mkoptions.py

```diff
@@ -835,6 +835,33 @@ def codegen_all_modules(modules, build_dir, dst_dir, tpl_options_h,
                 cond = ' || '.join(
                     ['name == "{}"'.format(x) for x in sorted(names)])
 
+                # Generate code for getOptionInfo
+                if option.name:
+                    constr = None
+                    fmt = {
+                        'type': option.type,
+                        'value': 'opts.{}.{}'.format(module.id, option.name),
+                        'default': option.default if option.default else '{}()'.format(option.type),
+                        'minimum': option.minimum if option.minimum else '{}',
+                        'maximum': option.maximum if option.maximum else '{}',
+                    }
+                    if option.type in ['bool', 'std::string']:
+                        constr = 'OptionInfo::ValueInfo<{type}>{{{default}, {value}}}'.format(**fmt)
+                    elif option.type == 'double' or is_numeric_cpp_type(option.type):
+                        constr = 'OptionInfo::NumberInfo<{type}>{{{default}, {value}, {minimum}, {maximum}}}'.format(**fmt)
+                    elif option.mode:
+                        values = ', '.join(map(lambda s: '"{}"'.format(s), option.mode.keys()))
+                        if not option.default:
+                            fmt['default'] = '???'
```

Comment:
Not sure I understand this line :)

Reply:
This line sets the default values for mode options (i.e. enums) that have no explicit default specified.

Reply:
When you set the default to `???`, when does it crash? Do we have mode options that do not define a default value? Do we even want to allow this?

Reply:
It will not crash. This information is communicated via the `ModeInfo` struct that holds the default value as `std::string`. We have `outputTag` that does not define a default... maybe it's a better idea to enforce defaults for mode options.

Reply:
Yeah `outputTag` is a bit special, but we could have a NONE tag for the default case and then enforce default values.

Reply:
After some offline discussion, we decided on having `NONE` but hide it from the generated help.